### DISCUSSION TITLE
[Swagger specification] skip the reference object property [OPENAPI 3.+]

### DISCUSSION THREAD

**Discussion by bmmule** [OAI/OpenAPI-Specification#2809] [Q&A]
Swagger specification : 

Consider in the case of "**allOf**" , i don't want to use one of the properties from my reference object , how should I achieve that?

**example :** 

```yaml
recipientAddress:
      allOf:
        - $ref: '#/components/schemas/Addresses'
Addresses:
      properties:
        addressCategory:
          type: string
          maxLength: 10
          description: Type of Address 
          enum: 
            - PERMANENT 
            - COMMUNICATON
          example: PERMANENT
        addressLine1:
          type: string
          maxLength: 40
          description: First line of the Address
          example: host tower
```
here i don't want property "**addressCategory**" in my object "**recipientAddress**".

**Comment by webron** [comment]
What's the point in cross posting here and https://github.com/OAI/OpenAPI-Specification/issues/2116#issuecomment-981813071? Also, when posting snippets, please make sure to use github's markdown otherwise it's hard to read your examples.

**Reply by webron** [reply]
You can't remove parts of an object when referencing it. If you want to reference only a part of it, you'd need to split `Addresses` into two objects and reference whichever part you need.

**Reply by bmmule** [reply]
In case my object has 20 properties and if i want to use 18 for other reference , i wanted to avoid duplication , is there any way to do it rather than having multiple copies?

**Reply by webron** [reply]
Separate the object into two, one with 18 properties, and the other with 2.

**Reply by bmmule** [reply]
ok , Thanks

**Comment by handrews** [comment]
answered, closing.
